src/libs/ReportUtils.ts

```diff
@@ -1227,7 +1227,8 @@ function isClosedExpenseReportWithNoExpenses(report: OnyxEntry<Report>): boolean
 /**
  * Whether the provided report is an archived room
  */
-function isArchivedRoom(report: OnyxInputOrEntry<Report> | EmptyObject, reportNameValuePairs?: OnyxInputOrEntry<ReportNameValuePairs> | EmptyObject): boolean {
+function isArchivedRoom(reportOrID: OnyxInputOrEntry<Report> | EmptyObject | string, reportNameValuePairs?: OnyxInputOrEntry<ReportNameValuePairs> | EmptyObject): boolean {
```

Comment:
I was looking at this one (and also `isInvoiceRoom` above) and I don't really like that the first param could be either a reportID (string) or a report (object). It makes the code feel much less uncertain and if I was looking at it, I'd be wondering when I should use a reportID vs just a report. Some suggestions I have:

- Keep it two separate parameters for clarity `(reportID, report)`
- Make two separate methods
- Anywhere that is only using the `reportID`, have it get the report from the collection first and then these methods would only have a `report` parameter.

Reply:
> I was looking at this one (and also isInvoiceRoom above) and I don't really like that the first param could be either a reportID (string) or a report (object). It makes the code feel much less uncertain and if I was looking at it, I'd be wondering when I should use a reportID vs just a report. Some suggestions I have:

It's the same way we use for other functions like `isMoneyRequestReport`, `isMoneyRequest`,...

Reply:
OK, I don't think that is a pattern that we should be following, so let's not do it here.

Reply:
@tgolen I updated with two separate methods.